main.tf

```diff
@@ -6,13 +6,13 @@ locals {
   configured_ami_image_id          = var.ami_image_id == null ? "" : var.ami_image_id
   need_ami_id                      = local.enabled ? local.features_require_ami && length(local.configured_ami_image_id) == 0 : false
   need_imds_settings               = var.metadata_http_endpoint != "enabled" || var.metadata_http_put_response_hop_limit != 1 || var.metadata_http_tokens != "optional"
-  features_require_launch_template = local.enabled ? length(var.resources_to_tag) > 0 || local.need_userdata || local.features_require_ami || local.need_imds_settings : false
+  features_require_launch_template = local.enabled ? length(var.resources_to_tag) > 0 || local.need_userdata || local.features_require_ami || local.need_imds_settings || length(var.additional_security_group_ids) : false
```

Comment:
I'm not sure if this is needed `|| length(var.additional_security_group_ids) `. Also it needs to be bool `length(var.additional_security_group_ids)  > 0`